### DISCUSSION TITLE
If the file is modified during debugging in Vscode, the execution line points to a strange place, is there a way to fix this?

### DISCUSSION THREAD

**Discussion by AAuhsoj** [microsoft/vscode-discussions#1166] [Extension Development QnA] [answered]
As in the title, I am using debug mode in vscode.

I'm running the code line by line and tracking the steps (pressing F11), as shown in the picture below, the highlights indicating the next run point to the strange code.

![화면 캡처 2024-05-03 221938](https://github.com/microsoft/vscode-discussions/assets/141807643/b8f8cac3-5f13-4c61-b5c5-1581c64014ee)

(Please understand that the image is not immediately visible due to the lack of reputation.)

My expectation is that the file will be modified in the middle of the run, and the debugger can't seem to reflect it. (To match the version, run 'pip uninstall ultralytics' before running debugger, which will be installed again in later code runs.)

My project is 'Multi Object Tracking, MOT' (reference below) https://github.com/mikel-brostrom/yolo_tracking

All you want is for the debugger to point to a proper line of execution.

I would appreciate it if you could tell me how to do this. (If there is any information that needs to be resolved, I will add it quickly.)

Save the file (ctrl + s), turn the file off and on.

(I didn't turn vscode off and on. I simply shut down the file at the top, turned it on again. (It doesn't affect execution))

**Comment by alefragnani** [comment, marked as answer]
Hi @AAuhsoj ,

It seems you are debugging Python, right? I'm not sure which extension you are using, maybe https://marketplace.visualstudio.com/items?itemName=ms-python.python, or some MOT specific but, in any case, the capability to update the debugger to reflect modifications in the source code, must be supported directly by the language extension, not VS Code. So, I would suggest you to report this on the extension repo, so the team responsible for the debugger could take a look, providing you guidance if there is some setting you must define, or plan the development of the feature for future releases.

By the way, for future feature requests about VS Code usage, please use the VS Code repository instead (https://github.com/microsoft/vscode/issues/new/choose). This Discussions repo is focused for extension development for now.

Hope this helps

**Comment by abrilurena** [comment]
Hi, @AAuhsoj! do you still need help with this?
